### PR TITLE
Use ARG --required

### PR DESCRIPTION
**Pull request by alexcb**
Updates Earthfiles to make use of the ARG --required flag.

Signed-off-by: Alex Couture-Beil <alex@earthly.dev>

### CODE REVIEW

File: Earthfile [hunk 1]
```diff
@@ -389,7 +389,7 @@ dind:
 dind-alpine:
     FROM docker:dind
     RUN apk add --update --no-cache docker-compose
-    ARG EARTHLY_TARGET_TAG_DOCKER
+    ARG --required EARTHLY_TARGET_TAG_DOCKER
```
Comment (vladaionescu) [inline Earthfile:392]: Isn't this a builtin arg?

Comment (alexcb) [reply Earthfile:392]: ah right! I wonder if this should cause an error in this case?

**Comment by vladaionescu** [comment]
BTW I think the release Earthfiles could benefit from some `--required`

File: Earthfile (hunk 1 above)
Comment (vladaionescu) [reply Earthfile:392]: Yeah, it might be a good idea

Comment (alexcb) [reply Earthfile:392]: I created https://github.com/earthly/earthly/issues/1625 to track this.

**Review by vladaionescu — APPROVED** [review]

**Pull request merged**